engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/StartEventSubscriptionManager.java

```diff
@@ -75,52 +81,61 @@ private void closeExistingMessageStartEventSubscriptions(
                 subscription.getRecord()));
   }
 
-  private DeployedProcess findLastMessageStartProcess(final ProcessMetadata processRecord) {
+  private DeployedProcess findLastStartProcess(
+      final ProcessMetadata processRecord, final Predicate<ExecutableCatchEventElement> predicate) {
```

Comment:
🔧 Let's clarify what is expected of the predicate by changing the name to `hasStartEventMatching`